ddtrace/tracer.py

```diff
@@ -411,10 +411,12 @@ def start_span(
             span = tracer.start_span('web.worker', child_of=context)
         """
         new_ctx = self._check_new_process()
+        if new_ctx:
+            child_of = new_ctx
 
         if child_of is not None:
             if isinstance(child_of, Context):
-                context = new_ctx or child_of
+                context = child_of
                 parent = child_of._get_current_span()
```

Comment:
The problem was this line. `child_of` would be the context coming from the parent process when forking which can contain strong references to spans which will be copied here, resulting in a leak.

The fix is to replace `child_of` entirely with `new_ctx` if it exists.

Reply:
Just trying to understand the logic here, if there is a parent process when forking we can safely ignore the `child_of` argument and instead use the new context 'cloned' from the active call context?

Reply:
Correct. However there is the possibility that `child_of` is a span and this happens right after a fork...

```python
span = tracer.trace(...)
pid = os.fork()

if pid == 0:
  tracer.start_span(child_of=span)
```

in which case this fix would break this behaviour...

Reply:
This has been addressed in e3c31a0